frontend/src/components/ViewActivities/activityfns.js

```diff
@@ -54,4 +57,30 @@ export function getField(activity, fieldName, defaultValue){
     return defaultValue;
   }
   return activity[fieldName];
+}
+
+/**
+ * Write contents into an activity in the database.
+ * 
+ * @param {string} tripId Database ID of the trip whose actiivty should be modified.
+ * @param {string} activityId Database ID of the activity to be modified.
+ * @param {Object} newValues Dictionary of the new values in {fieldName: newValue} form
+ */
+export async function writeActivity(tripId, activityId, newValues) {
+  // todo: check if tripId or activityId is not valid. (#58)
+  if (newValues === null) {
+    return false;
+  }
+
+  const act = db.collection(DB.COLLECTION_TRIPS).doc(tripId)
+    .collection(DB.COLLECTION_ACTIVITIES).doc(activityId);
+  
+  try {
+    // Note: newValues cannot contain lists. Check documentation for update().
+    const a = await act.update(newValues);
+    return a ? true : false;
+  } catch (e) {
+    console.log(e);
+    return false;
```

Comment:
Just curious, why return a promise holding a boolean rather than just returning the result of the update?

Reply:
That does return the boolean. `a ? true: false` returns `true` if `a` exists and `false` otherwise.